src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -6403,6 +6403,24 @@ void AuraEffect::HandleBattlegroundPlayerPosition(AuraApplication const* aurApp,
         bg->RemovePlayerPosition(target->GetGUID());
 }
 
+// Increase armor by <AuraEffect.BasePoints> % of your <primary stat>
+void AuraEffect::HandleModArmorPctFromPrimaryStat(AuraApplication const* aurApp, uint8 mode, bool apply) const
+{
+    if (!(mode & (AURA_EFFECT_HANDLE_CHANGE_AMOUNT_MASK | AURA_EFFECT_HANDLE_STAT)))
+        return;
+
+    // only players have primary stats
+    Player* player = aurApp->GetTarget()->ToPlayer();
+    if (!player)
+        return;
+
+    Stats primaryStat = player->GetPrimaryStats();
+    int32 armorAmount = CalculatePct(player->GetStat(primaryStat), GetAmount());
+    if (apply)
+        for (uint8 x = SPELL_SCHOOL_NORMAL; x < MAX_SPELL_SCHOOL; ++x)
```

Comment:
This implementation is incorrect/incomplete for two reasons
* Armor is UNIT_MOD_ARMOR, not UNIT_MOD_RESISTANCE_*
* Armor bonus must update dynamically when stats change - requires calling UpdateArmor() in Player::UpdateStats and moving this calculation inside UpdateArmor() (StatSystem.cpp)
* The stat to use is  `MiscValueA != -2 ? Stats(MiscValueA) : player->GetPrimaryStats()`

Reply:
Those are 3 reasons

Reply:
Verify I did it correctly, but know tihs:

The client doesn't seem to get the updated armor. When looking on the character window, with equipment and stats, armor is always 5, even when ironfur is on and it adds around 180 armor.

I also verified the armor is being added in the server by debugging an attack.

Reply:
Not good, it should be showing up on client

Reply:
Found the issue, sort of. Seems like the client can only see the `Resistances` and not the `BonusResistanceMods`.
Also, specifically with `IronFur` which has this effect it will update the `Resistances`.

I've changed the PR to reflect that, so that `SPELL_AURA_MOD_ARMOR_PCT_FROM_STAT` will now affect baseValue rather than totalValue.